template-tags.php

```diff
@@ -417,19 +417,41 @@ function coauthors_ids( $between = null, $betweenLast = null, $before = null, $a
 	), null, $echo );
 }
 
-function get_the_coauthor_meta( $field ) {
+/**
+ * Outputs the co-authors Meta Data
+ *
+ * @param string $field Required The user field to retrieve.[login, email, nicename, display_name, url, type]
+ */
+function get_the_coauthor_meta( $field ) { 
 	global $wp_query, $post;
 
 	$coauthors = get_coauthors();
 	$meta = array();
-
-	foreach ( $coauthors as $coauthor ) {
+        
+        if ( in_array( $field, array( 'login', 'pass', 'nicename', 'email', 'url', 'registered', 'activation_key', 'status' ) ) )
+                $field = 'user_' . $field;
+        
+	foreach ( $coauthors as $coauthor ) {                
 		$user_id = $coauthor->ID;
-		$meta[ $user_id ] = get_the_author_meta( $field, $user_id );
-	}
+                
+                if ( isset( $coauthor->type )  && 'user_url' === $field ) {
+                        if ( 'guest-author' === $coauthor->type) {
+                                $field = 'website';
+                        } 
+                } else if ( 'website' === $field ) {
```

Comment:
Hi! Are you able to explain the reasoning behind this block?  If we called `get_the_coauthor_meta( 'website' )`,  guest authors would yield nothing seeing as their property is called `'website'` and we would be looking for `'user_url'`.  We should only set `'website'` to `'user_url'` if it is not a guest-author type or if the type is not set.

Reply:
@rebeccahum : Are we allowing to call get_the_coauthor_meta( 'website' )? As per my knowledge, the function originally allowed only 'url' as the field parameter for both, guest_authors as well as wp_users. So, I have updated the function accordingly.

Reply:
@nilzari Based on your line of code, it is doing a check to see if it's passed in...so, based on that -- the only reason we would ever need to perform a check for `$field` equating to `'website'` is if we did a call to  `get_the_coauthor_meta( 'website' )`.
However, with that said, I don't think we need `'website'` to be passed in because I'm not seeing it in the list here for `get_the_author_meta` here: https://developer.wordpress.org/reference/functions/get_the_author_meta/ I'd presume we'd want `get_the_coauthor_meta` to mirror the core function as closely as possible.

Reply:
@rebeccahum : Below is the reason for condition ``` } else if ( 'website' === $field ) { ``` .

If the  ```$coauthors``` array has two objects, first one for guest_author and second one for wp_user (since the post has two authors). Now we make a call to ```get_the_coauthor_meta( 'url' )```. In this case, since the first object is guest_author, the ```$field``` is updated to 'website'. For the second object which is wp_user, the function will check if the ```$field``` is 'website', it will again change it to 'user_url'.

Reply:
@nilzari Ah okay, I see what you are trying to do now.  I don't think it will work as expected because in this case it's _actually_ checking to see if `'website'` was passed into `$field` once `isset( $coauthor->type )  && 'user_url' === $field` evaluates to `false`.  These lines you have here will be enough to do what you want:
```
if ( isset( $coauthor->type )  && 'user_url' === $field ) {
                        if ( 'guest-author' === $coauthor->type) {
                                $field = 'website';
                        } 
```
Because in the scenario you are describing, in the first one for guest_author, it will set the `$field` value to `'website'` and grab that property and then for a wp_user, it doesn't need to set anything since the `$field` value already at `user_url`.

Reply:
@rebeccahum : In the first iteration, for first object 'guest_author', the  ```$field``` values is set to 'website' by below lines of code .

```
if ( isset( $coauthor->type )  && 'user_url' === $field ) {
                        if ( 'guest-author' === $coauthor->type) {
                                $field = 'website';
                        } 
```

When the code will check for second object in the second iteration, it will find ```$field``` value as 'website' which is not applicable for wp_user. So I have to again set the value for ```$field``` as 'user_url'. I have to explicitly change the ```$field``` value for wp_user in this case since it is updated in first iteration and doesn't set back to 'user_url' unless i change it explicitly. Therefore, below condition is necessary.

```
else if ( 'website' === $field ) {
                        $field = 'user_url';
                }
```

Reply:
@nilzari Ahhh now I see the edge case that you are trying to account for.  I understand now, good work.  Carry on!